### DISCUSSION TITLE
Server component rendering error

### DISCUSSION THREAD

**Discussion by neo87cs** [vercel/next.js#71144] [Ideas]
### Goals

1. Give the possibility to intercept server rendering errors and log in external services

### Non-Goals

1. Override the existing informations obfuscation on client side

### Background

Currently if an error happens in a server component during rendering phase, the error is sent to the ErrorBoundary declared in `error.{jsx|tsx}` file in a anonymized way to avoid sensible informations exposition to the browser. While this is awesome and we have a digest to connect an Error with its counterpart in server logs, it is time spending and require one of these two actions to be made:
- Centralized log if there are more than one server machine executing the NextJS server, imagine fetching log from n servers to find the digest inside;
- Check the server log to find the digest (ideally one server).
Some companies has third parties logging tools (in our case Rollbar), we cannot log something significant in server side to understand what is going on.
The only alternative i've found is https://github.com/infodusha/next-rsc-error-handler but while it works (someone quoted it in https://github.com/trpc/trpc/issues/5682, the dependency could lead to unwanted behaviors in future.

### Proposal

Implement a mechanism to provide a callback or permits to handle an event somehow, in a pure server side, to log the error in an external provider. It is similar to what usually backends does and could helps a lot on fixing stuffs fastly.
I am interested in contributing on it but i didn't see the NextJS codebase yet.

**Comment by unstubbable** [comment]
Take a look at [`onRequestError`](https://nextjs.org/docs/app/api-reference/file-conventions/instrumentation#onrequesterror-optional). This might fit your needs.

**Reply by neo87cs** [reply]
It is available starting from v15? I was going to proxy the `console.error` in the server to intercept it

**Reply by unstubbable** [reply]
Yes, see https://nextjs.org/docs/app/api-reference/file-conventions/instrumentation#version-history
